File: frontend/types/rate-override.ts
export interface RateOverride {
  id: number
  roomTypeCode: string
  roomTypeName: string
  name: string
  startDate: string
  endDate: string
  price: number
  weekdays: number[] | null
  priority: number
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface RoomTypeRateOverrideCreateRequest {
  name: string
  startDate: string
  endDate: string
  price: number
  weekdays: number[] | null
  priority: number
}

export interface RateOverrideUpdateRequest extends RoomTypeRateOverrideCreateRequest {
  roomTypeId: number
}

File: backend/src/main/java/com/example/hotelmanagement/entity/RateOverride.java
package com.example.hotelmanagement.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "rate_overrides")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RateOverride extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_type_id")
    private RoomType roomType;

    @Column(nullable = false, length = 120)
    private String name;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    @Column(nullable = false, precision = 14, scale = 2)
    private BigDecimal price;

    @Column(columnDefinition = "JSON")
    private String weekdays;

    @Column(nullable = false)
    @Builder.Default
    private Integer priority = 0;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;
}
